calico_node/startup/startup.go

```diff
@@ -687,35 +723,35 @@ func checkConflictingNodes(client *client.Client, node *api.Node) {
 		// addresses have changed.  If so warn the user as it could be
 		// an indication of multiple nodes using the same name.  This
 		// is not an error condition as the IPs could actually change.
-		if theirNode.Metadata.Name == node.Metadata.Name {
-			if theirIPv4 != nil && ourIPv4 != nil && !theirIPv4.IP.Equal(ourIPv4.IP) {
+		if theirNode.Name == node.Name {
+			if theirIPv4 != "" && ourIPv4 != "" && theirIPv4 != ourIPv4 {
```

Comment:
The IP address is actually a CIDR - so we probably need to parse the CIDRs here and compare the IP portion of the CIDR.

Reply:
Not sure I understand the need to parse the CIDR here. If there's a conflict, shouldn't they have the same `IPvXAddress` string with subnet and all?

Reply:
well theoretically they could have the same IP and different subnet
 -if it's misconfigured.... which is after all what we're testing for